server/auctionapp/src/main/java/com/atlantbh/auctionapp/service/AuthService.java

```diff
@@ -98,4 +117,65 @@ public void authenticate(String email, String password) throws Exception {
         }
     }
 
+    public String updateResetPasswordToken(String email) throws MessagingException, UnsupportedEncodingException {
+        UserEntity userEntity = userRepository.findByEmail(email);
+        String token = RandomString.make(30);
+        String url = allowedURL + "/reset-password/" + token;
+        if (userEntity != null) {
+            userEntity.setResetPasswordToken(token);
+            userEntity.setResetPasswordTokenCreatedAt(LocalDateTime.now());
+            userRepository.save(userEntity);
+            sendEmail(email, url);
+        } else {
+            logger.error("User with email: " + email + " does not exist.");
+            throw new NotFoundException("User with email: " + email + " does not exist.");
+        }
+
+        return "Please check your email for the reset password link.";
+    }
+
+    public String updatePassword(ResetPasswordRequest resetPasswordRequest) {
+        UserEntity userEntity = userRepository.findByResetPasswordToken(resetPasswordRequest.getToken());
+        if (userEntity == null) {
+            logger.error("User with token: " + resetPasswordRequest.getToken() + " does not exist.");
+            throw new NotFoundException("User with token: " + resetPasswordRequest.getToken() + " does not exist.");
+        }
+        if (!LocalDateTime.now().isBefore(userEntity.getResetPasswordTokenCreatedAt().plusMinutes(30))) {
```

Comment:
Would it be easier to read if it is LocalDateTime.now().isAfter(...) than having negated isBefore()?